### ISSUE TITLE
Conflicts between imgsync and pbr

### ISSUE THREAD

**Issue by NeolithEra** [alvarolopez/imgsync#2]
Hi, users are unable to run **_imgsync_** due to dependency conflict with _**pbr**_ package. As shown in the following full dependency graph of **_imgsync_**, **_imgsync_** requires _**pbr <1.0,>=0.6**_，while **_python-glanceclient_** requires _**pbr >=2.0.0**_.

According to pip’s _“first found wins”_ installation strategy, _**pbr 0.11.1**_ is the actually installed version. However, _**pbr 0.11.1**_ does not satisfy _**>=2.0.0**_.

### Dependency tree-----------
```
imgsync - 1.1.3
| +- backports-lzma(install version:0.0.14 version range:*)
| +- dateutils(install version:0.6.7 version range:*)
| | +- argparse(install version:1.4.0 version range:*)
| | +- python-dateutil(install version:2.8.1 version range:*)
| | +- pytz(install version:2019.3 version range:*)
| +- keystoneauth1(install version:3.18.0 version range:*)
| | +- iso8601(install version:0.1.12 version range:>=0.1.11)
| | +- os-service-types(install version:1.7.0 version range:>=1.2.0)
| | | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | +- requests(install version:2.22.0 version range:>=2.14.2)
| | | +- certifi(install version:2019.11.28 version range:>=2017.4.17)
| | | +- chardet(install version:3.0.4 version range:<3.1.0,>=3.0.2)
| | | +- idna(install version:2.8 version range:>=2.5,<2.9)
| | | +- urllib3(install version:1.25.7 version range:<1.26,>=1.21.1)
| | +- six(install version:1.13.0 version range:>=1.10.0)
| | +- stevedore(install version:1.31.0 version range:>=1.20.0)
| | | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | | +- six(install version:1.13.0 version range:>=1.10.0)
| +- oslo-config(install version:6.12.0 version range:*)
| | +- debtcollector(install version:1.22.0 version range:>=1.2.0)
| | | +- funcsigs(install version:1.0.2 version range:>=1.0.0)
| | | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | | +- six(install version:1.13.0 version range:>=1.10.0)
| | | +- wrapt(install version:1.11.2 version range:>=1.7.0)
| | +- enum34(install version:1.1.6 version range:>=1.0.4)
| | +- netaddr(install version:0.7.19 version range:>=0.7.18)
| | +- oslo.i18n(install version:3.25.0 version range:>=3.15.3)
| | +- pyyaml(install version:5.2 version range:>=3.12)
| | +- requests(install version:2.22.0 version range:>=2.18.0)
| | | +- certifi(install version:2019.11.28 version range:>=2017.4.17)
| | | +- chardet(install version:3.0.4 version range:<3.1.0,>=3.0.2)
| | | +- idna(install version:2.8 version range:>=2.5,<2.9)
| | | +- urllib3(install version:1.25.7 version range:<1.26,>=1.21.1)
| | +- rfc3986(install version:1.3.2 version range:>=1.2.0)
| | +- six(install version:1.13.0 version range:>=1.10.0)
| | +- stevedore(install version:1.31.0 version range:>=1.20.0)
| | | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | | +- six(install version:1.13.0 version range:>=1.10.0)
| +- oslo-log(install version:3.45.0 version range:*)
| | +- debtcollector(install version:1.22.0 version range:>=1.19.0)
| | | +- funcsigs(install version:1.0.2 version range:>=1.0.0)
| | | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | | +- six(install version:1.13.0 version range:>=1.10.0)
| | | +- wrapt(install version:1.11.2 version range:>=1.7.0)
| | +- monotonic(install version:1.5 version range:>=1.4)
| | +- oslo.config(install version:6.12.0 version range:>=5.2.0)
| | +- oslo.context(install version:2.23.0 version range:>=2.20.0)
| | +- oslo.i18n(install version:3.25.0 version range:>=3.20.0)
| | +- oslo.serialization(install version:2.29.2 version range:>=2.25.0)
| | +- oslo.utils(install version:3.42.1 version range:>=3.36.0)
| | +- pbr(install version:5.4.4 version range:>=3.1.1)
| | +- pyinotify(install version:0.9.6 version range:>=0.9.6)
| | +- python-dateutil(install version:2.8.1 version range:>=2.7.0)
| | +- six(install version:1.13.0 version range:>=1.11.0)
| +- pbr(install version:0.11.1 version range:<1.0,>=0.6)
| | +- pip(install version:19.3.1 version range:*)
| +- python-glanceclient(install version:2.17.0 version range:*)
| | +- keystoneauth1(install version:3.18.0 version range:>=3.6.2)
| | | +- iso8601(install version:0.1.12 version range:>=0.1.11)
| | | +- os-service-types(install version:1.7.0 version range:>=1.2.0)
| | | | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | | +- requests(install version:2.22.0 version range:>=2.14.2)
| | | | +- certifi(install version:2019.11.28 version range:>=2017.4.17)
| | | | +- chardet(install version:3.0.4 version range:<3.1.0,>=3.0.2)
| | | | +- idna(install version:2.8 version range:>=2.5,<2.9)
| | | | +- urllib3(install version:1.25.7 version range:<1.26,>=1.21.1)
| | | +- six(install version:1.13.0 version range:>=1.10.0)
| | | +- stevedore(install version:1.31.0 version range:>=1.20.0)
| | | | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | | | +- six(install version:1.13.0 version range:>=1.10.0)
| | +- oslo.i18n(install version:3.25.0 version range:>=3.15.3)
| | +- oslo.utils(install version:3.42.1 version range:>=3.33.0)
| | +- pbr(install version:5.4.4 version range:>=2.0.0)
| | +- prettytable(install version:0.7.2 version range:<0.8,>=0.7.1)
| | +- pyopenssl(install version:19.1.0 version range:>=17.1.0)
| | | +- cryptography(install version:2.8 version range:>=2.8)
| | | +- six(install version:1.13.0 version range:>=1.5.2)
| | +- requests(install version:2.22.0 version range:>=2.14.2)
| | | +- certifi(install version:2019.11.28 version range:>=2017.4.17)
| | | +- chardet(install version:3.0.4 version range:<3.1.0,>=3.0.2)
| | | +- idna(install version:2.8 version range:>=2.5,<2.9)
| | | +- urllib3(install version:1.25.7 version range:<1.26,>=1.21.1)
| | +- six(install version:1.13.0 version range:>=1.10.0)
| | +- warlock(install version:1.3.3 version range:>=1.2.0,<2)
| | | +- jsonpatch(install version:1.24 version range:<2,>=0.10)
| | | | +- jsonpointer(install version:2.0 version range:>=1.9)
| | | +- jsonschema(install version:3.2.0 version range:>=0.7,<4)
| | | +- six(install version:1.13.0 version range:*)
| | +- wrapt(install version:1.11.2 version range:>=1.7.0)
| +- requests(install version:2.22.0 version range:*)
| | +- certifi(install version:2019.11.28 version range:>=2017.4.17)
| | +- chardet(install version:3.0.4 version range:<3.1.0,>=3.0.2)
| | +- idna(install version:2.8 version range:>=2.5,<2.9)
| | +- urllib3(install version:1.25.7 version range:<1.26,>=1.21.1)
| +- six(install version:1.13.0 version range:*)
```

Thanks for your help.
Best,
Neolith

**Comment by NeolithEra** [comment]
### Suggested Solution

1.	Fix your direct dependency to be **_pbr>=0.6_**. I have checked this revision will not affect your downstream projects now.
2.	Remove your direct dependency **_pbr_**, and use **_pbr_** transitively introduced by **_python-glanceclient_**.

@alvarolopez Which solution do you prefer, 1 or 2?
Please let me know your choice. I can submit a PR to solve this issue.

**Comment by alvarolopez** [comment]
Hi @NeolithEra 

I think that the best way is (1).

Can you make the PR?
